### PR TITLE
Create new maintenance window (no repeat)

### PR DESCRIPTION
**Pull request by MuhammadKhalilzadeh**
Doing small commits:
A part of `create new maintenance window (no repeat)`

So far this is the screenshot:

Screenshot 2024-08-01 230614

For components like Date Picker and Time Picker, as mentioned before, so far these are the built-in MUI components I'm using for now.

### CODE REVIEW

**Comment by MuhammadKhalilzadeh** [comment]
@ajhollid 
I've only worked with 3 files, but I not sure why I have 11 files changed! Why is that?

**Comment by gorkem-bwl** [comment]
Looks good so far. I am not sure about the gap between each line. Can you check the settings page Daniel implemented and copy it from there?

**Comment by MuhammadKhalilzadeh** [comment]
@gorkem-bwl 
Yeah, there are some differences in gaps, but we'll be fixed in the end.
Sure, I've check the settings 👍🏻

File: Client/src/Components/CreateMaintenanceWindow/index.jsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
+import { Box, Stack, Typography } from "@mui/material";
```
Comment (ajhollid) [inline Client/src/Components/CreateMaintenanceWindow/index.jsx:1]: This is a Page, not a Component, so it should be loctaed at `src/Pages/Maintenance/CreateMaintenanceWindow/index.jsx`

File: Client/src/Components/CreateMaintenanceWindow/index.jsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+const directory = {
+  title: "Create a maintenance window",
+  description: "Your pings won’t be sent in this time frame.",
+};
+
+const repeatOptions = [
+  { name: "Don't repeat", _id: 1 },
+  { name: "Repeat daily", _id: 2 },
+  { name: "Repeat weekly", _id: 3 },
```
Comment (ajhollid) [inline Client/src/Components/CreateMaintenanceWindow/index.jsx:21]: Convention is genreally that the unique identifier should be the first key in an object if it is a map type object.

Unfamiliar users will be confused by this and may asusme that `name` is the unique identifier rathern than `_id`

```
{_id: 1, name: "Don't repeat}"
```
is clearer and doesn't leave room for error

File: README.md [hunk 3]
```diff
@@ -6,48 +6,45 @@
 ![](https://img.shields.io/github/issues-pr/bluewave-labs/bluewave-uptime)
 ![](https://img.shields.io/github/issues/bluewave-labs/bluewave-uptime)
```
Comment (ajhollid) [inline README.md:8]: Next time please make a separate pull request for updating the readme.

A pull request should be very focused and only deal with one feature / fix / update at a time.

Imagine we need to undo this README change for whatever reason.  If it was a separate PR we could just revert that PR.

Now since this PR has more than one feature / fix our job is more difficult.

**Review by ajhollid — CHANGES_REQUESTED** [review]
There are a couple of minor issues that would be good to address, please see code review for details.

My biggest concern is that I do not understand why this pull request has 51 commits including commits from months ago.

A pull request should be _very_ focused and only deal with exactly one feature / fix / update at a time and should not contain any commits that are not related to that feature / fix / update

If you could close this PR and create a new clean PR with only the commits related to this feature that would be great :+1:

File: README.md (hunk 3 above)
Comment (MuhammadKhalilzadeh) [reply README.md:8]: @ajhollid 
This is not my modification!
I just fetched, and this among changed files!
I have only added and modified only and only 3 files:

`Client/src/Components/CreateMaintenanceWindow/index.jsx`
`Client/src/Components/CreateMaintenanceWindow/index.css`
`App.js`

These are the only files I've worked on. Don't know why other files are added. I've just made a sync on the branch, and I guess other files are added through that!

Comment (ajhollid) [reply README.md:8]: The issue is probably that you've merged `master` in to your feature branch:

image

This is not something that we should ever do, we should only ever merge `develop` into `feature`.

If you merge `master` into `feature` and then `feature` into `develop` then you have effectively merged `master` in to `develop`, which is something we never want to do.

The fix here is simple, just take your chagnes and move them to a new branch that you branch off of `develop`.

1.  `git checkout develop`
2.  `git checkout -b <my_feature>`
3.  Make your changes, commit, and push

This will ensure a clean git commit history and will make sure you don't have unwanted changes in your code :+1:
